scipy/stats/stats.py

```diff
@@ -6731,13 +6731,16 @@ def ks_2samp(data1, data2, alternative='two-sided', mode='auto'):
 
     if mode == 'asymp':
         # The product n1*n2 is large.  Use Smirnov's asymptoptic formula.
+        # Ensure float to avoid overflow in multiplication
+        # sorted because the one-sided formula is not symmetric in n1, n2
+        m, n = sorted(np.array([n1, n2], float), reverse=True)
```

Comment:
No need to create a numpy array here. This is quite a bit faster:
```suggestion
        m, n = sorted([float(n1), float(n2)], reverse=True)
```

Reply:
Done.